File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/retrievemultipleconditionoperatorslinq1.cs
// <snippetretrievemultipleconditionoperatorslinq1>


 var contacts = (from c in svcContext.ContactSet
                 join a in svcContext.AccountSet on c.ParentCustomerId.Id equals a.AccountId
                 where (a.Name == "Litware, Inc.")
                 where (c.Address1_StateOrProvince == "WA"
                 && (c.Address1_Telephone1.StartsWith("(206)") ||
                     c.Address1_Telephone1.StartsWith("(425)"))
                 && (c.Address1_City == "Redmond" ||
                     c.Address1_City == "Bellevue" ||
                     c.Address1_City == "Kirkland" ||
                     c.Address1_City == "Seattle")
                 && (c.EMailAddress1 != null && c.EMailAddress1 != ""))
                 select new Contact
                 {
                  ContactId = c.ContactId,
                  FirstName = c.FirstName,
                  LastName = c.LastName,
                  Address1_Telephone1 = c.Address1_Telephone1
                 });

 // Display the results.
 Console.WriteLine("List all contacts matching specified parameters");
 Console.WriteLine("===============================================");
 foreach (Contact contact in contacts)
 {
  Console.WriteLine("Contact ID: {0}", contact.Id);
  Console.WriteLine("Contact Name: {0}", contact.FullName);
  Console.WriteLine("Contact Phone: {0}", contact.Address1_Telephone1);
 }
 Console.WriteLine("<End of Listing>");
 Console.WriteLine();
 #endregion

// </snippetretrievemultipleconditionoperatorslinq1>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/retrievemultipleconditionoperatorslinq1.vb
' <snippetretrievemultipleconditionoperatorslinq1>


 Dim contacts = ( _
     From c In orgContext.CreateQuery(Of Contact)() _
     Join a In orgContext.CreateQuery(Of Account)() _
     On c.ParentCustomerId.Id Equals a.AccountId _
     Where (a.Name.Equals("Litware, Inc.")) _
     Where (c.Address1_StateOrProvince.Equals("WA") _
            AndAlso (c.Address1_Telephone1.StartsWith("(206)") _
             OrElse c.Address1_Telephone1.StartsWith("(425)")) _
         AndAlso (c.Address1_City.Equals("Redmond") _
                  OrElse c.Address1_City.Equals("Bellevue") _
                  OrElse c.Address1_City.Equals("Kirkland") _
                  OrElse c.Address1_City.Equals("Seattle")) _
              AndAlso (c.EMailAddress1 IsNot Nothing _
                       AndAlso c.EMailAddress1 IsNot "")) _
     Select New Contact With
            {
                .ContactId = c.ContactId,
                .FirstName = c.FirstName,
                .LastName = c.LastName,
                .Address1_Telephone1 = c.Address1_Telephone1
            })

 ' Display the results.
 Console.WriteLine("List all contacts matching specified parameters")
 Console.WriteLine("===============================================")
 For Each contact As Contact In contacts
     Console.WriteLine("Contact ID: {0}", contact.Id)
     Console.WriteLine("Contact Name: {0}", contact.FullName)
     Console.WriteLine("Contact Phone: {0}",
                       contact.Address1_Telephone1)
 Next contact
 Console.WriteLine("<End of Listing>")
 Console.WriteLine()

' </snippetretrievemultipleconditionoperatorslinq1>